src/lib/tecs/TECS.cpp

```diff
@@ -627,8 +641,26 @@ void TECSControl::resetIntegrals()
 	_throttle_integ_state = 0.0f;
 }
 
+void TECS::initControlParams(float target_climbrate, float target_sinkrate, float eas_to_tas, float pitch_limit_max,
+			     float pitch_limit_min, float throttle_min, float throttle_setpoint_max, float throttle_trim)
+{
+	// Update parameters from input
+	// Reference model
+	_reference_param.target_climbrate = target_climbrate;
+	_reference_param.target_sinkrate = target_sinkrate;
+	// Control
+	_control_param.tas_min = eas_to_tas * _equivalent_airspeed_min;
+	_control_param.pitch_max = pitch_limit_max;
+	_control_param.pitch_min = pitch_limit_min;
+	_control_param.throttle_trim = throttle_trim;
+	_control_param.throttle_max = throttle_setpoint_max;
+	_control_param.throttle_min = throttle_min;
+}
+
 void TECS::initialize(const float altitude, const float altitude_rate, const float equivalent_airspeed,
-		      const float eas_to_tas)
+		      float eas_to_tas, float throttle_min, float throttle_setpoint_max,
+		      float throttle_trim, float pitch_limit_min, float pitch_limit_max, float target_climbrate,
+		      float target_sinkrate)
```

Comment:
I think this is just an old remnant of something you tried? Those are not used in the function since it is now only used internally and the initControlParam is set beforehand.

Reply:
0e0aee1766082766be83f5a0c11feffd97ee492c